openssl/src/symm.rs

```diff
@@ -601,6 +650,12 @@ pub fn encrypt_aead(
 ) -> Result<Vec<u8>, ErrorStack> {
     let mut c = Crypter::new(t, Mode::Encrypt, key, iv)?;
     let mut out = vec![0; data.len() + t.block_size()];
+
+    if t.is_ccm() {
```

Comment:
Will other authenticated suites like GCM break if we always do these?

Reply:
Sadly, yes. OCB mode (when supported) will also [need special handling](https://www.openssl.org/docs/man1.1.0/crypto/EVP_CipherUpdate.html#GCM-and-OCB-Modes).

Reply:
Ugh